### PR TITLE
allow opting out of recursively unloaded of modules via recursive_module_unload easyconfig parameter

### PR DESCRIPTION
**Pull request by boegel**
cfr. #3627

### CODE REVIEW

File: test/framework/options.py [hunk 1]
```diff
@@ -1974,10 +1974,15 @@ def test_recursive_module_unload(self):
...
+
             if get_module_syntax() == 'Lua':
                 toy_module += '.lua'
+                is_loaded_regex = re.compile(r'if not \( isloaded\("gompi/2018a"\) \)', re.M)
+            else:
+                # Tcl syntax
+                is_loaded_regex = re.compile(r"if { !\[is-loaded gompi/2018a\] }", re.M)
+
             toy_module_txt = read_file(toy_module)
-            is_loaded_regex = re.compile(r"if { !\[is-loaded gompi/2018a\] }", re.M)
```
Comment (migueldiascosta) [inline test/framework/options.py:1980]: so, with `Lua` this test wasn't actually testing? (just checking if I understand)

Comment (boegel) [reply test/framework/options.py:1980]: Indeed. We were checking for a missing pattern using a Tcl regex pattern in a Lua module file.

That probably shows this test should be rewritten a bit, but I just wanted to fix the obvious problem when I noticed it.

**Review by migueldiascosta — APPROVED** [review]
lgtm

**Comment by migueldiascosta** [comment]
Going in, thanks @boegel!

**Pull request merged**